firecrown/ccl/sources/sources.py

```diff
@@ -374,3 +374,84 @@ def render(self, cosmo, params, systematics=None):
             systematics[systematic].apply(cosmo, params, self)
 
         self.dndz_interp_ = Akima1DInterpolator(self.z_, self.dndz_)
+
+
+class CMBLSource(Source):
+    """A galaxy cluster source.
+
+    Parameters
+    ----------
+    sacc_tracer : str
+        The name of the tracer in the SACC file.
+    scale : float, optional
+        The default scale for this source. Usually the default of 1.0 is
+        correct.
+    systematics : list of str, optional
+        A list of the source-level systematics to apply to the source. The
+        default of `None` implies no systematics.
+
+    Attributes
+    ----------
+    ell_orig : np.ndarray
+        The original ells used in the sacc tracer. Set after the call
+        to `read`.
+    beam_orig : np.ndarray
```

Comment:
Why are you keeping `ell` and the `beam` here? They are not used iiuic.

Reply:
So I hoped to keep a copy of them in case they are changed after calling render. So does it make sense that I keep them here?

Reply:
But what are they used for in the code? What systematic or otherwise would need a copy of `ell` which specifies where things are measured?

Reply:
Yeah I think it's not needed.

Reply:
But for `beam` does it make sense to read it here and make a copy in `render()`? Or should I directly read and copy it in `render()`?

Reply:
the first. Read it here and then make a copy on render